openlibrary/plugins/openlibrary/api.py

```diff
@@ -142,7 +142,7 @@ def POST(self, work_id):
         from openlibrary.core.models import Bookshelves
 
         user = accounts.get_current_user()
-        i = web.input(edition_id=None, action="add", redir=False, bookshelf_id=None)
+        i = web.input(edition_id=None, action="add", redir=False, bookshelf_id=None, dont_remove = False)
```

Comment:
We may want to add a docstring my assumption is false;

We can't use the action because we *do* want to add the book if it doesn't exist yet. If the book already exists, we don't want to remove it.